src/ssl_stubs.c

```diff
@@ -262,7 +263,41 @@ CAMLprim value ocaml_ssl_get_error_string(value unit)
   CAMLreturn(caml_copy_string(buf));
 }
 
+CAMLprim value ocaml_ssl_error_struct(value err_func)
+{
+  CAMLparam1(err_func);
+  CAMLlocal3(result, libstring, reasonstring);
+  unsigned long code = 0;
+  switch(Int_val(err_func)) {
+    case 0:
+      code = ERR_get_error();
+      break;
+    case 1:
+      code = ERR_peek_error();
+      break;
+    case 2:
+      code = ERR_peek_last_error();
+      break;
+  }
+  result = caml_alloc_tuple(3);
+  const char *lib = ERR_lib_error_string(code);
+  const char *reason = ERR_reason_error_string(code);
+  if (lib != NULL) {
+    libstring = caml_copy_string(lib);
+  } else {
+    libstring = caml_copy_string("lib(0)");
```

Comment:
how about making `lib` and `reason` `string option`s instead of making up these strings?